### PR TITLE
Stellar wip

### PR DESCRIPTION
**Pull request by mliu**
https://github.com/user-attachments/assets/237d2408-9935-40cc-805d-99d820c663dc

### CODE REVIEW

File: packages/huma-shared/src/utils/config.ts [hunk 1]
```diff
@@ -82,6 +82,8 @@ const getEthereumDappUrl = (isDev = false) =>
   `https://${getDevPrefix(isDev)}app.huma.finance`
 const getSolanaDappUrl = (isDev = false) =>
   `https://${getDevPrefix(isDev)}solapp.huma.finance`
+const getStellarDappUrl = (isDev = false) =>
```
Comment (shan-57blocks) [inline packages/huma-shared/src/utils/config.ts:85]: I don't think we need this anymore

File: packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  requestAccessFn: () => void
+}
+
+export const StellarConnectionContext =
+  createContext<StellarConnectionContextType>({
+    address: '',
+    requestAccessFn: () => {},
+  })
+
+const Watcher = new WatchWalletChanges(1000)
```
Comment (shan-57blocks) [inline packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx:15]: What does 1000 mean? One second?

Comment (shan-57blocks) [reply packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx:15]: Could you please explain why we need this watcher?

File: packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      if (watcherResults.address !== address) {
+        setAddress(watcherResults.address)
+      }
+      // document.querySelector('#network').innerHTML = watcherResults.network
+    })
+    // Only watch once
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const requestAccessFn = async () => {
```
Comment (shan-57blocks) [inline packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx:35]: Could you please add comments for this function?

File: packages/huma-web-shared/src/stellar/hooks/useTokenBalance.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react'
+import { StellarChainInfo } from '@huma-finance/shared'
+import { fetchTokenBalance } from '../utils'
+
+export const useTokenBalance = (
```
Comment (shan-57blocks) [inline packages/huma-web-shared/src/stellar/hooks/useTokenBalance.ts:5]: Can we rename to useStellarTokenBalance?

File: packages/huma-web-shared/src/stellar/utils/fetchTokenBalance.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  SorobanRpc,
+  Contract,
+  Address,
+  scValToNative,
+  TransactionBuilder,
+  BASE_FEE,
+  Account,
+} from '@stellar/stellar-sdk'
+
+export async function fetchTokenBalance(
```
Comment (shan-57blocks) [inline packages/huma-web-shared/src/stellar/utils/fetchTokenBalance.ts:12]: ditto

**Review by shan-57blocks — DISMISSED** [review]

File: packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx (hunk 2 above)
Comment (mliu) [reply packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx:15]: Will do

File: packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx (hunk 3 above)
Comment (mliu) [reply packages/huma-web-shared/src/stellar/StellarWeb3Provider.tsx:35]: Will do

**Review by PlayJok3r — APPROVED** [review]

**Pull request merged**